web/client/plugins/TOC/components/Legend.jsx

```diff
@@ -132,9 +135,13 @@ class Legend extends React.Component {
     validateImg = (img) => {
         // GeoServer response is a 1x2 px size when legend is not available.
         // In this case we need to show the "Legend Not available" message
-        if (img.height <= 1 && img.width <= 2) {
+        const imgError = img.height <= 1 && img.width <= 2;
+        if (imgError) {
             this.onImgError();
         }
+        if ((this.props.layer.dynamicLegendIsEmpty ?? null) !== imgError) {
+            this.props.onUpdateNode({ dynamicLegendIsEmpty: imgError });
+        }
```

Comment:
In https://github.com/allyoucanmap/MapStore2/commit/a11b6736dd0cc5175977ec75a212d3c7d205d309 same as previous comment